components/blocks/carousel.tsx

```diff
@@ -55,20 +61,35 @@ export const Carousel = ({ data }) => {
           renderIndicator={createCarouselIndicator}
         >
           {data.items &&
-            data.items.map((props, index: React.Key) =>
-              createCarouselItemImage(props, index, data)
-            )}
+            data.items.map((item, index) => (
+              <CarouselItemImage
+                key={index + item.label}
+                imgSrc={item.imgSrc}
+                label={item.label}
+                index={index}
+                carouselSchema={item.carouselSchema}
+              />
+            ))}
         </CarouselImplementation>
       </Container>
     </Section>
   );
 };
 
-const createCarouselItemImage = (props, index: React.Key, carouselSchema) => {
-  const { imgSrc, label } = props;
+type CarouselItemImageProps = {
+  imgSrc: string;
+  label: string;
+  index: number;
+  // tinacms accepts any, workaround for better type safety for this component
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  carouselSchema: any;
```

Comment:
maybe we should define some kind of type there to reduce the extra variables coming through.